nchelpers/__init__.py

```diff
@@ -308,15 +368,30 @@ def metadata(self):
         return self.UnifiedMetadata(self)
 
     @property
-    def is_unprocessed_model_output(self):
-        """True iff the content of the file is unprocessed model output.
-        This allows us to discern between raw amd downscaled model output, for example"""
-        try:
-            self.metadata.model
-        except AttributeError:
-            return False
-        else:
-            return True
+    def is_unprocessed_gcm_output(self):
+        """True iff the content of the file is unprocessed GCM output."""
+        return self.product == 'output'
+
+    @property
+    def is_downscaling_output(self):
+        """True iff the content of the file is downscaling output."""
+        return self.product == 'downscaling output'
+
+    @property
+    def is_hydromodel_output(self):
+        """True iff the content of the file is hydrological model output of any kind."""
+        return self.product == 'hydrological model output'
+
+    @property
+    def is_hydromodel_dgcm_output(self):
+        """True iff the content of the file is output of a hydrological model driven by downscaled GCM data."""
+        return self.is_hydromodel_output and hasattr(self, 'downscaling_method_id') and hasattr(self, 'driving_model_id')
+
+    @property
+    def is_hydromodel_iobs_output(self):
+        """True iff the content of the file is output of a hydrological model driven by interpolated observational data."""
```

Comment:
I'm waiting on input from Arelia about these models and their metadata. When I've got that I'll add tests for that type of file too.